unison-cli/tests/Unison/Test/LSP.hs

```diff
@@ -69,6 +79,55 @@ term = This
 |],
         True,
         Right (Type.Ref (Reference.unsafeFromText "#6kbe32g06nqg93cqub6ohqc4ql4o49ntgnunifds0t75qre6lacnbsr3evn8bkivj68ecbvmhkbak4dbg4fqertcpgb396rmo34tnh0"))
+      ),
+      ( "Test annotations within bindings for do-block elements",
+        [here|
+term = do
+  first = false
+  second = tr^ue
+  first && second
+        |],
+        True,
+        Left (Term.Boolean True)
+      ),
+      ( "Test annotations within bindings for let-block elements",
+        [here|
+term = let
+  first = false
+  second = tr^ue
+  first && second
+        |],
+        True,
+        Left (Term.Boolean True)
+      ),
+      ( "Test annotations within actions for let-block elements",
+        [here|
+term = let
+  first = false
+  first && tr^ue
+        |],
+        True,
+        Left (Term.Boolean True)
+      ),
+      -- ( "Test annotations for blocks with destructuring binds",
```

Comment:
This one was getting long, so I'll defer fixing annotations for destructuring binds to a different PR.